Source/Core/DolphinQt2/GameList/GameFile.h

```diff
@@ -65,18 +67,95 @@ class GameFile final
   QString GetLongName(DiscIO::Language lang) const { return m_long_names[lang]; }
+  QMap<DiscIO::Language, QString> GetLongNames() { return m_long_names; }
   QString GetLongMaker(DiscIO::Language lang) const { return m_long_makers[lang]; }
+  QMap<DiscIO::Language, QString> GetLongMakers() { return m_long_makers; }
   QString GetDescription(DiscIO::Language lang) const { return m_descriptions[lang]; }
+  QMap<DiscIO::Language, QString> GetDescriptions() { return m_descriptions; }
   bool Install();
   bool Uninstall();
   bool ExportWiiSave();
 
+  class Cache
+  {
+  public:
+    void Update(GameFile* gamefile);
+    void Save();
+    void Load();
+    bool IsCached(const QString& path);
+
+    QDateTime GetTimestamp(const QString& path);
+    u64 GetTitleID(const QString& path);
+    QString GetGameID(const QString& path);
+    QString GetMakerID(const QString& path);
+
+    QString GetMaker(const QString& path);
+    QMap<DiscIO::Language, QString> GetLongMakers(const QString& path);
+    QMap<DiscIO::Language, QString> GetShortMakers(const QString& path);
+
+    QString GetInternalName(const QString& path);
+    QMap<DiscIO::Language, QString> GetLongNames(const QString& path);
+    QMap<DiscIO::Language, QString> GetShortNames(const QString& path);
+
+    DiscIO::Platform GetPlatform(const QString& path);
+    DiscIO::Region GetRegion(const QString& path);
+    DiscIO::Country GetCountry(const QString& path);
+    DiscIO::BlobType GetBlobType(const QString& path);
+
+    u64 GetRawSize(const QString& path);
+    QMap<DiscIO::Language, QString> GetDescriptions(const QString& path);
+    u16 GetRevision(const QString& path);
+    u8 GetDiscNumber(const QString& path);
+    int GetRating(const QString& path);
+    QString GetApploaderDate(const QString& path);
+    QPixmap GetBanner(const QString& path);
+
+    static Cache& Get();
+
+  private:
+    static Cache instance;
+    Cache();
+
+    QFile file;
+
+    QMap<QString, QDateTime> m_timestamp;
+
+    // IDs
+    QMap<QString, qint64> m_title_id;
+    QMap<QString, QString> m_game_id;
+    QMap<QString, QString> m_maker_id;
+
+    // Maker
+    QMap<QString, QString> m_maker;
+    QMap<QString, QMap<qint32, QString>> m_maker_long;
+    QMap<QString, QMap<qint32, QString>> m_maker_short;
+
+    // Name
+    QMap<QString, QString> m_name_internal;
+    QMap<QString, QMap<qint32, QString>> m_name_long;
+    QMap<QString, QMap<qint32, QString>> m_name_short;
+
+    // Other
+    QMap<QString, qint32> m_platform;
+    QMap<QString, qint32> m_region;
+    QMap<QString, qint32> m_country;
+    QMap<QString, qint32> m_blob_type;
+    QMap<QString, quint16> m_revision;
+    QMap<QString, quint32> m_raw_size;
+    QMap<QString, QMap<qint32, QString>> m_description;
+    QMap<QString, quint8> m_disc_number;
+    QMap<QString, qint32> m_rating;
+    QMap<QString, QString> m_apploader_date;
+    QMap<QString, QPixmap> m_banner;
+  };
```

Comment:
Can't we have a single `QMap` that maps to one object for each game, so that we won't have to do as many map lookups?